sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -538,12 +538,12 @@ public static SharedKeyCredential getSharedKeyCredential(HttpPipeline httpPipeli
                 long count = pos + blockSize > length ? length - pos : blockSize;
                 byte[] cache = new byte[(int) count];
                 int lastIndex = data.read(cache);
-                currentTotalLength[0] += lastIndex;
-                if (lastIndex < count) {
+                if (lastIndex == -1 && currentTotalLength[0] < length) {
```

Comment:
This is getting closer to an implementation that would handle a stream that is buffering slowly but there is an issue that this will have random chunks of data missing. With the current implementation given a scenario where I have 5 100 byte chunks I could have this happen:

Read 100
Read 100
Read 10 (slow buffering)
Read 10 (slow buffering)
Read 100
Throw UnexpectedLengthException, expected 500 got 320

But there could still be more data loading into the stream.

Instead there should be a looping mechanism on each concatMap emissions where if I read that 10 I try to read more until one of the following scenarios

1. I have filled by current buffer, wrap this and trigger the next concatMap emission
2. I have hit end of file, wrap this and trigger on complete
3. I have hit end of file and I got less than expected, trigger an error

Now the scenario should look like

Read 100
Read 100,
Read 10 (slow buffering), read 10 (slow buffering), read 80
Read 100
Read 100
Everything is good here, send the data to the service.